src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -352,11 +352,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 boundArguments.Add(boundArgument)
 
                 Dim elementType = GetTupleFieldType(boundArgument, argumentSyntax, diagnostics, hasErrors)
-                elementTypes.Add(elementType)
-
                 If elementType Is Nothing Then
                     hasNaturalType = False
+                    elementType = ReclassifyExpression(boundArgument, diagnostics).Type
```

Comment:
Agree. Alternatively I would need to hang a binder on the node and reclassify on demand.